fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/rdf/JcrRdfTools.java

```diff
@@ -410,33 +405,20 @@ private void findOrCreateHashUri(final IdentifierConverter<Resource, FedoraResou
     }
 
     private Resource getSkolemizedResource(final IdentifierConverter<Resource, FedoraResource> idTranslator,
-                                           final RDFNode resource) throws RepositoryException {
+                                           final RDFNode resource, final String topic) throws RepositoryException {
         final AnonId id = resource.asResource().getId();
 
         if (!skolemizedBnodeMap.containsKey(id)) {
-            jcrTools.findOrCreateNode(session, skolemizedPrefix());
-            final String pid = pidMinter.get();
-            final String path = skolemizedPrefix() + pid;
-            final Node preexistingNode = getClosestExistingAncestor(session, path);
-            final Node orCreateNode = jcrTools.findOrCreateNode(session, path);
-            orCreateNode.addMixin(FEDORA_SKOLEM);
-
-            if (preexistingNode != null) {
-                AbstractService.tagHierarchyWithPairtreeMixin(preexistingNode,
-                        orCreateNode);
-            }
-
-            final Resource skolemizedSubject = nodeToResource(idTranslator).convert(orCreateNode);
+            final String base = idTranslator.asString(createResource(topic));
+            final Resource skolemizedSubject = idTranslator.toDomain(base + "#" + blankNodeIdentifier());
+            findOrCreateHashUri(idTranslator, skolemizedSubject);
             skolemizedBnodeMap.put(id, skolemizedSubject);
         }
 
         return skolemizedBnodeMap.get(id);
     }
 
-    private static String skolemizedPrefix() {
-        return "/.well-known/genid/";
+    private String blankNodeIdentifier() {
+        return "genid" + randomUUID().toString();
```

Comment:
Managing a shared counter seems like overkill — I'm fine with UUIDs, but it would be nice if we could have shorter ids.  Though given the use of UUIDs elsewhere, users should be accustomed to them by now.